readthedocs/search/parsers.py

```diff
@@ -20,6 +20,7 @@ def __init__(self, version):
         self.project = self.version.project
         self.storage = build_media_storage
 
+    @functools.cache
     def _get_page_content(self, page):
```

Comment:
I'm afraid this may implode our memory usage :D, some options:

- Just read the content from storage twice.
- When parsing the content (search indexing), we can calculate the hash of the main content, and save that for later, so we will be saving just the hash, and not the whole content. This will make things faster, but it mixes things...

Reply:
Went with the second idea in https://github.com/readthedocs/readthedocs.org/pull/11646/commits/ed5377d9fe0b6b27269cbaec5c3d590226c6420b